test/IntegrationTests/test_listfile.py

```diff
@@ -0,0 +1,293 @@
+
+        logger.info("Successfully tested file listing when no files have been uploaded")
+
+    def test_list_files_from_empty_bucket(self):
+        logger.info("Testing file listing for an empty bucket")
+
+        #Create a new customer to get a valid GUID
+        add_customer_url=f"{self.BASE_URL}/addcustomer"
+        response=requests.post(add_customer_url)
+        self.assertEqual(response.status_code, 200, "Failed to create customer")
+
+        customer_data=response.json()
+        customer_guid=customer_data.get("customer_guid")
+        logger.info(f"Created new customer with GUID: {customer_guid}")
+
+        #Verify the bucket exists but no files are uploaded
+        list_files_url=f"{self.BASE_URL}/listfiles"
+        params={"customer_guid": customer_guid}
+        response=requests.get(list_files_url, params=params)
+
+        #Verify the HTTP response status code
+        if response.status_code==200:
+            logger.info("Response status code is valid and matches the expected value (200). Bucket is empty.")
+        else:
+            logger.error(f"Unexpected status code: {response.status_code}. Expected: 200.")
+            raise AssertionError(f"Expected status code 200 but got {response.status_code}")
+
+        #Assert the response is successful and the file list is empty
+        self.assertEqual(response.status_code, 200, "Failed to list files for empty bucket")
+        self.assertEqual(response.json().get("files"), [], "Expected no files in the bucket")
+
+        logger.info("Successfully tested file listing for an empty bucket")
+
+    def test_list_files_from_invalid_customer_guid(self):
+        logger.info("Executing test_list_files_from_invalid_customer_guid: Testing list files with an invalid customer_guid")
+
+        invalid_customer_guid="invalid-customer-guid"
+
+        url=f"{self.BASE_URL}/listfiles"
+        logger.info(f"Sending GET request to {url} with invalid customer_guid")
+
+        #Request parameters
+        params={"customer_guid": invalid_customer_guid}
+
+        #Make GET request
+        response=requests.get(url, params=params)
+        logger.info(f"Received response status code: {response.status_code} for URL: {url}")
+        logger.info(f"Response content: {response.text}")
+
+        self.assertIn(response.status_code, [404], f"Expected status code 404 but got {response.status_code}")
+        data=response.json()
+
+        #Verify the response contains the 'detail' field
+        self.assertIn("detail", data, "'detail' not found in response data")
+        logger.info("'detail' found in response data, verifying its content")
+
+        #Verify that the error message indicates invalid customer_guid
+        self.assertEqual(data["detail"], "Invalid customer_guid provided", "Unexpected error message content")
+
+        logger.info("Test completed successfully for test_list_files_from_invalid_customer_guid")
+
+    def test_list_files_two_files_uploaded(self):
```

Comment:
Do we think that these two testcases are the same functionality wise?

 def test_list_files_from_valid_customer(self):
 def test_list_files_two_files_uploaded(self):
 
 If yes should we remove one of them?